djangocms_versioning/managers.py

```diff
@@ -78,12 +78,68 @@ def current_content(self, **kwargs):
             .values_list("vers_pk", flat=True)
         return qs.filter(versions__pk__in=pk_filter)
 
+    def latest_content(self, **kwargs):
+        """Returns the "latest" content object which is in this order
+           1. a draft version (should it exist)
+           2. a published version (should it exist)
+           3. any other version with the highest pk
+
+        This filter assumes that there can only be one draft created and that the draft as
+        the highest pk of all versions (should it exist).
+        """
+        """
+        While this is probably to most general approach it does not work for MySql :-(
```

Comment:
There is some logic in the core that does a db check, we could use that and if for any reason it's fixed in mysqly in the future, it could be removed: https://github.com/django-cms/django-cms/blob/042c11c6341d288987843307089339ba8247b69a/cms/models/pluginmodel.py#L76

Reply:
I've removed the more general approach since it also seems to be by a factor of 3 slower than the implemented version.